Platform/CommonBoardPkg/Library/TccLib/TccLib.c

```diff
@@ -0,0 +1,639 @@
+  @retval EFI_NOT_FOUND           Unable to locate PTCM Binary
+*/
+EFI_STATUS
+EFIAPI
+GetPtctPtcmBinary (
+  IN  UINT32            PtctIndex,
+  OUT PTCT_HEADER_ENTRY **PtctEntry
+  )
+{
+  PTCT_PTCM_BINARY                *PtctPtcmBinary;
+
+  if ((PtctEntry ==NULL) || (*PtctEntry != NULL) || (mTccInfoHob == NULL) || (PtctIndex >= 1)) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  // Skip making PTCM Binary PTCT entry if PTCM binary doesnt exist
+  if ((mTccInfoHob->Data.PtcmBinaryAddress == 0) || (mTccInfoHob->Data.PtcmBinarySize == 0)) {
+    DEBUG ((DEBUG_INFO, "PtcmBinary does not exist\n"));
+    return EFI_NOT_FOUND;
+  }
+
+  PtctPtcmBinary = AllocatePool (sizeof (PTCT_PTCM_BINARY));
+  if (PtctPtcmBinary == NULL) {
+    return EFI_OUT_OF_RESOURCES;
+  }
+
+  PtctPtcmBinary->Header.Size = sizeof (PTCT_PTCM_BINARY);
+  PtctPtcmBinary->Header.Type = PtctPtcmBinaryType;
+  PtctPtcmBinary->Header.Version = PTCT_PTCM_BINARY_VERSION;
+
+  PtctPtcmBinary->Address = mTccInfoHob->Data.PtcmBinaryAddress;
+  PtctPtcmBinary->Size = mTccInfoHob->Data.PtcmBinarySize;
+
+  *PtctEntry = (PTCT_HEADER_ENTRY *)PtctPtcmBinary;
+
+  DEBUG ((DEBUG_INFO, "PtcmBinary[%d] Address: 0x%X,\n", PtctIndex, PtctPtcmBinary->Address));
+  DEBUG ((DEBUG_INFO, "PtctPtcmBinary->Size: 0x%X\n", PtctPtcmBinary->Size));
+  return EFI_SUCCESS;
+}
+
+const PtctCallEntry mPtctCallTable[] = {
+  {GetPtctPtcdLimits, 1},
+  {GetPtctGtL3WayMasks, 1},
+  {GetPtctBuffer, MAX_TCC_BUFFERS},
+  {GetPtctMemHierarchyLatency, MAX_MEMORY_HIERARCHY_LATENCY_ENTRIES},
+  {GetPtctPtcmBinary, 1}
+};
+
+/**
+  Update PTCT  (Platform Tuning Configuration Table) ACPI Table
+
+  @param[in] EFI_ACPI_DESCRIPTION_HEADER   PTCT ACPI Table entry
+
+  @retval EFI_STATUS            Status returned from PTCT Table entries
+          EFI_NOT_FOUND         Loader Global Pointer not found
+                                TccHob not found
+          EFI_INVALID_PARAMETER Incorrect Ptct table entries
+**/
+EFI_STATUS
+EFIAPI
+UpdateAcpiPtctTable (
+    IN EFI_ACPI_DESCRIPTION_HEADER *PtctTable
+  )
+{
+  UINT8                         *DestAddr;
+  EFI_STATUS                    Status;
+  PTCT_HEADER_ENTRY             **PtctList;
+  PTCT_HEADER_ENTRY             *PtctEntry;
+  UINT32                        PtctTableLength;
+  UINT32                        PtctCallTableCount;
+  UINT32                        Index;
+  UINT32                        MaxPtctEntries;
+  UINT32                        PtctEntryListIndex;
+  UINT32                        PtctCallIndex;
+  LOADER_GLOBAL_DATA            *LdrGlobal;
+
+  Status = EFI_SUCCESS;
+  if (PtctTable == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  LdrGlobal   = (LOADER_GLOBAL_DATA *)GetLoaderGlobalDataPointer();
+  if (LdrGlobal == NULL) {
+    return EFI_NOT_FOUND;
+  }
+
+  mTccInfoHob = GetNextGuidHob (&gTccInfoHobGuid, LdrGlobal->FspHobList);
+  if (mTccInfoHob == NULL) {
+    return EFI_NOT_FOUND;
+  }

```

Comment:
Can we do this in the platform code? Depending on platform, GUID of Hob structure may vary.